docs/html/user_guide.rst

```diff
@@ -289,7 +289,10 @@ organisation and use that everywhere. If the thing being installed requires
 "helloworld" to be installed, your fixed version specified in your constraints
 file will be used.
 
-Constraints file support was added in pip 7.1.
+Constraints file support was added in pip 7.1. In :ref:`Resolver
+changes 2020` we did a fairly comprehensive overhaul and stripped
+constraints files down to being purely a way to specify global
+(version) limits for packages.
```

Comment:
I think it might be worth noting that the removed features were never (as far as we could determine) part of the original design requirements for constraints, and were essentially (undocumented) quirks of the previous implementation.

(I don't want this to sound like we're being defensive here, but I *do* think it's worth including a little context rather than just baldly saying "we ripped out a load of functionality").

Reply:
@pfmoore Sounds good! :+1: I am grateful for your context and will add it.